pandas/plotting/_core.py

```diff
@@ -1706,21 +1706,37 @@ def _plot(data, x=None, y=None, subplots=False,
         plot_obj = klass(data, subplots=subplots, ax=ax, kind=kind, **kwds)
     else:
         if isinstance(data, ABCDataFrame):
+            new_data = data.copy()  # don't modify until necessary
```

Comment:
this is not necessary, just use 'data' don't add something else here

Reply:
I added the copy to fix the bug from #20056 so we get correct integer indexing - what did you have in mind?

Reply:
I think it's best to convert from integer to labels as soon as possible. Can you check if you can do it at the very start of `_plot`? Then you don't have to worry about this.